### ISSUE TITLE
updating primary key Value failed

### ISSUE THREAD

**Issue by Sysix** [vlucas/spot2#210]
Hey guys,

I think I have found a bug. 
When I update a Entity's primary key, it would be no changes at all. 
My fields are: 
```php
    public static function fields()
    {
        return [
            'scope' => [
                'type' => 'string',
                'primary' => true
            ],
            'is_default' => [
                'type' => 'boolean',
                'default' => 0
            ]
        ];
    }
```

My Update code is:

```php
            /** @var \Spot\EntityInterface $entity */
            $entity->data(array(
'scope' => $newScope
));

            $scopes->getMapper()->update($entity);
```

Xdebug Outputs on Doctrine\DBAL\Connection on line 665:

```php
$sql = "UPDATE `oauth_scopes` SET `scope` = ? WHERE `scope` = ?"
$params = array('newValue', 'newValue'); 
```

But it should be: 

```php
$params = array('newValue', 'oldValue'); 
```

**Comment by tuupola** [comment]
Primary keys should not contain business logic. Better to use `unsigned integer` as primary key. You can still have scope as `unique`.

**Comment by Sysix** [comment]
Hello,

yes I know its the "better" art of building a Database. But this way should be valid too. With "Vanilla-SQL" it works.  So I think is should work with spot2 too.

I added a unsigned primary key `id` and now it works. But this problem is still active for the framework.

**Comment by vlucas** [comment]
@Sysix I acknowledge that you *can* do this with raw SQL, but in general, it's never a good idea to design a schema without a primary key. The database engine makes all sorts of optimizations off of it, and any ORM will have to have some kind of unique reference to the current record.

Since you have not defined any `primary key` OR `unique` index, "bad things™" can happen if an ORM tries to guess at what you are doing just by using all the columns and current values. Things like: 

- accidentally editing multiple records that match the criteria (multiple records with the same values for `scope` and `is_default`)
- Causing slow query performance - Since there is no primary or unique key, an ORM like Spot would be forced to send ALL columns and ALL values to any `UPDATE` and `DELETE` queries in the `WHERE`, which would run much slower than using a single primary key.

In short, while an ORM like Spot *could* do this, it would be solving for a case that would never be optimized or performant, and is widely considered to be bad practice in schema design anyways. This won't be fixed in Spot.
